src/shared/gpt.c

```diff
@@ -4,16 +4,6 @@
 #include "string-util.h"
 #include "utf8.h"
 
-/* Gently push people towards defining GPT type UUIDs for all architectures we know */
-#if !defined(GPT_ROOT_NATIVE) ||                                        \
-        !defined(GPT_ROOT_NATIVE_VERITY) ||                             \
-        !defined(GPT_ROOT_NATIVE_VERITY_SIG) ||                         \
-        !defined(GPT_USR_NATIVE) ||                                     \
-        !defined(GPT_USR_NATIVE_VERITY) ||                              \
-        !defined(GPT_USR_NATIVE_VERITY_SIG)
-#pragma message "Please define GPT partition types for your architecture."
-#endif
```

Comment:
hmm? please see:

 https://github.com/systemd/systemd/pull/21290/commits/1fb2d8fcb69bcdbab0a5dd23bbf02f729e47e656

for the reasons i added this.

Reply:
i.e. the idea is that RootImage=, portable services, systemd-dissect, systemd-nspawn should should just work on every arch, and not just the ones that also do UEFI. the concept of GPT, portable services, containers are all arch independent, hence we should support them on all archs.

Yes, systed-gpt-auto-generator doesn't make too much sense on archs where the rootfs is not on GPT, but all other tools that parse GPT can certainly tkae benefit of it.